pkg/controller/secret/secret_controller.go

```diff
@@ -105,9 +143,19 @@ type ReconcileSecret struct {
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *ReconcileSecret) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
-	reqLogger.Info("Reconciling " + userDataSecret)
 
-	validUserData, err := secrets.GenerateUserData()
+	privateKey, err := secrets.GetPrivateKey(request.NamespacedName, r.client)
```

Comment:
Don't pass client around, if possible create a struct around it make them methods, so that you can share

Reply:
Are you suggesting moving GetPrivateKey() out of the secrets library? That would be the only way to stop passing the client around

Reply:
Nope, I meant don't pass the client to every method. Create a struct called 
```suggestion
type secretConfig struct {
   privateKey []byte
   secret *corev1.Secret
   clientSet
}
func NewSecretConfig(clientSet) *SecretConfig {
// Construct Secret Config
}
     // Call NewSecretConfig()
      privateKey, err := secretConfig.GetPrivateKey(request.NamespacedName, r.client)
```
With this association becomes easier and would be easier to map secret to a privateKey. You don't have to implement this.

Reply:
I think this adds unnecessary abstraction, we still would have to create a new secretClient every time. I'd rather leave it as it is for now